tests/main/snap-quota-journal/task.yaml

```diff
@@ -0,0 +1,96 @@
+# requires systemd v245+
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -centos-8-*
+  - -debian-10-64
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+  - -ubuntu-18.04-*
+  - -ubuntu-core-18-*
+
+prepare: |
+  snap pack test-snapd-journal-quota
+  snap install --dangerous test-snapd-journal-quota_1.0_all.snap
+  tests.cleanup defer snap remove --purge test-snapd-journal-quota
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  echo "Starting service and verifying that log messages are sent to default journal namespace"
+  snap start test-snapd-journal-quota.logger
+
+  # Now ask for logs and lets see if we can see messages from 'LogDaemon'
+  # The format of the log messages are 'LogDaemon: message number: X'
+  retry -n 5 sh -c "$TESTSTOOLS/journal-state get-log | MATCH LogDaemon"
+
+  # Create a journal quota group and set arbitrary limits, we will then verify these are written
+  # correctly for journald
+  echo "Create a group with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-journal-quota
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services test-snapd-journal-quota.logger | MATCH "test-snapd-journal-quota.logger\s+enabled\s+active"
+
+  # Verify the contents of /etc/systemd/journald@snap-group-one.conf
+  echo "Verifying the journal config contents"
+  MATCH "^SystemMaxUse=16000000"        < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RuntimeMaxUse=16000000"       < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitIntervalSec=10000us" < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitBurst=100"           < /etc/systemd/journald@snap-group-one.conf
+  
+  # Now ask for the namespace logs and lets see if we can see messages from 'LogDaemon'
+  retry -n 5 sh -c "$TESTSTOOLS/journal-state get-log --namespace snap-group-one | MATCH LogDaemon"
+
+  # Change the size and ensure that the journal is restarted
+  echo "Updating size of journal to ensure size changes"
+  snap set-quota group-one --journal-size=64MB
+
+  # Ensure that the new size is printed in the journal logs. We requested 64MB but
+  # this will most likely print out as 61.0M, so lets be a bit flexible
+  "$TESTSTOOLS"/journal-state get-log --namespace snap-group-one | MATCH "max 6[0-9].[0-9]M,"
+
+  # Ensure that the unit file is correct
+  echo "Ensuring the unit file is correct"
+  systemctl show --property=LogNamespace snap.test-snapd-journal-quota.logger.service | MATCH "LogNamespace=snap-group-one"
+  LOGGER_PID=$(systemctl show --property=MainPID --value snap.test-snapd-journal-quota.logger.service)
```

Comment:
I would move this line near the beginning of the test, as here it looks like it's related to the "Ensure that the unit file is correct" step.

Reply:
Or maybe after the `echo` line which introduces the next step.

Reply:
Good suggestion, thank you for your review while I was gone!